src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -1,69 +1,86 @@
+     * components, their corresponding policy violations/vulnerabilities and declaration locations.
      * @param rapidFullResults
-     * @param scanOutputFolder
-     * @return
+     * @param scanOutputFolder Detect's output subdirectory where this file will be saved
+     * @param sourceDir source directory of project being scanned
      * @throws DetectUserFriendlyException
      */
-    public static File locateComponentsforNonPersistentOnlineDetectorScan(List<DeveloperScansScanView> rapidFullResults, File scanOutputFolder) throws DetectUserFriendlyException {
-        // In Part II:
-            // given a rapid scan full result response, call ScanResultToCLLComponentTransformer to get CLL input file (components-source.json)
-            // call library w/ CLL input
-            // return the resulting file (not anticipating any post-processing except saving it in the correct directory)
-        return generatePlaceHolderJsonFileForNow(scanOutputFolder);
-    }
-
-    public static File locateComponentsForOfflineDetectorScan(BdioResult bdio, File scanOutputFolder) throws DetectUserFriendlyException {
-        // In Part II:
-            // given a BDIO, call BdioToCLLComponentTransformer to get CLL input file (components-source.json)
-            // call library w/ CLL input
-            // return the resulting file (not anticipating any post-processing except saving it in the correct directory)
-        return generatePlaceHolderJsonFileForNow(scanOutputFolder);
+    public void locateComponentsForNonPersistentOnlineDetectorScan(List<DeveloperScansScanView> rapidFullResults, File scanOutputFolder, File sourceDir) {
+        List<Component> componentsList = scanResultTransformer.transformScanResultToComponentList(rapidFullResults);
+        Input componentLocatorInput = generateComponentLocatorInput(componentsList, sourceDir);
+        callComponentLocatorObfuscatedJar(componentLocatorInput, scanOutputFolder);
     }
 
     /**
-     * Placeholder file for testing purposes, saves a JSON file in the appropriate directory with the appropriate name.
-     * @param scanOutputFolder
-     * @return
+     * Given a BDIO, generates an output file consisting of the list of unique components detected and their declaration
+     * locations.
+     * @param bdio from running offline Detector scan
+     * @param scanOutputFolder Detect's output subdirectory where this file will be saved
+     * @param sourceDir source directory of project being scanned
      * @throws DetectUserFriendlyException
      */
-    private static File generatePlaceHolderJsonFileForNow(File scanOutputFolder) throws DetectUserFriendlyException {
+    public void locateComponentsForOfflineDetectorScan(BdioResult bdio, File scanOutputFolder, File sourceDir) {
+        List<Component> componentsList = bdioTransformer.transformBdioToComponentList(bdio);
+        Input componentLocatorInput = generateComponentLocatorInput(componentsList, sourceDir);
```

Comment:
common code for this and locateComponentsForNonPersistentOnlineDetectorScan could potentially be extracted to a method unless you expect things to deviate in the future.